Zendesk WIP: Migrating to sdk v2

oguzkocer opened the pull request:
This PR integrates the v2 of Zendesk SDK. This version of their SDK is much better and in hindsight, I am glad we made the upgrade, I just hope there won't be any issues with it.

The goal of the PR is to replicate the exact same behavior we had with v1. Updating the SDK has brought up some more opportunities to improve the code, but I decided not to do any of that in this PR to keep in concise. Here is their documentation that could prove useful during the review: https://developer.zendesk.com/embeddables/docs/android-support-sdk/nutshell

**To test:**

Since this PR contains an SDK change, we need to do a through testing of everything related to Zendesk. Reading the description and test steps of #7739 could be useful especially if you're unfamiliar with the Zendesk work so far.

Although it'd be good to do so, I don't think it's necessary to test every caller of ZendeskHelper for this PR. It might be best to just go to Me -> Help & Support and try the `Contact us`, `Help Center` & `My Tickets`. Once you verify that it's working, a logged out test would be good. So, logout and then go to login page, tap on the help button and try to create a ticket.

Please ping me for the Zendesk credentials if you don't already have them.

/cc @aerych @ScoutHarris (just to keep in the loop, they are not reviewing)

Here are updated screenshots:

![screenshot_1527886473](https://user-images.githubusercontent.com/662023/40863077-9dff30ee-65bc-11e8-83b8-77088379d8d2.png)
![screenshot_1527886480](https://user-images.githubusercontent.com/662023/40863078-9e202e52-65bc-11e8-9ade-e3389a9988e2.png)
![screenshot_1527886488](https://user-images.githubusercontent.com/662023/40863079-9e2fcf74-65bc-11e8-8b74-91698e182953.png)
![screenshot_1527886505](https://user-images.githubusercontent.com/662023/40863080-9e3e83e8-65bc-11e8-9788-90b8305671e6.png)

oguzkocer commented on WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt:
```diff
@@ -82,22 +70,24 @@ fun showZendeskHelpCenter(
-        configureZendesk(context, supportEmail, supportName, accountStore, siteStore, selectedSite)
+        zendeskInstance.setIdentity(zendeskIdentity(supportEmail, supportName))
     } else {
         zendeskInstance.setIdentity(zendeskIdentity(null, null))
```
Yeap, same thing. We can use the `AnonymousIdentity` directly if you prefer. I like one function handling the identity creation and other code not knowing about it though.

0nko replied:
Either is fine 👍.

oguzkocer commented on WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt:
```diff
@@ -162,17 +159,6 @@ private fun configureZendesk(
-            }
-        }
-
 private fun zendeskIdentity(email: String?, name: String?): Identity =
```
Same as above - I don't think I understand the issue of the naming here as this one literally creates and returns a Zendesk identity. We can't call it `getZendeskIdentity` because that'll mean that we have an identity to return and not create it. If you think it's better we can call it `createZendeskIdentity`.

Also, I think it'd be more productive and faster if you suggest a naming at least to explain what you find wrong with it.

0nko replied:
Yes, `createZendeskIdentity()` is what I had in mind.

oguzkocer commented on WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt:
```diff
@@ -130,29 +121,35 @@ fun showAllTickets(
 // Helpers
 
-private fun configureZendesk(
+private fun zendeskConfig(
```
> This also applies to other functions in ZendeskHelper

I think it'd be better if you just comment on them specifically with suggestions on what would be better.

I think we can add `get`, `create` type of prefixes to some of them (although I don't think it'll make it better in most cases), but I don't agree that there is anything else wrong with them. I think you prefer names such as `configureThis` or `doThat` but the functions you commented on so far don't do anything but create and return a value. It's the callers job to take action with them and they are named accordingly as well.

oguzkocer replied:
Just to make sure there are no misunderstandings, I am both trying to better understand your feedback and explain the parts of the current implementation that might not have been clear to figure out the best approach of the naming.

I am definitely not disregarding your feedback. It hopefully didn't come off that way anyway, but I am kind of paranoid in these type of situations and wanted to be extra sure. Thanks again for the review.

0nko replied:
No worries :). So the problem I have with non-verb method naming is that it's not immediately clear whether it sets or gets something and you have to look. For example, `zendeskConfig(...)`. Does it create a new Zendesk config? Then `createZendeskConfig()` or `buildZendeskConfig()` would work. Does it just store the supplied values somewhere? (`storeZendeskConfig()`) Does it just retrieve an existing instance (`getZendeskConfig()`)? 

I guess it's not as obvious when you write the code yourself since you know what you meant by it. I'm not trying to be petty about the names. It's just when I see a function named like an object I always have to pause for a moment and think what it does. That's all :).

oguzkocer commented:
@0nko Thanks for the review. I left replies to your comments and it should be ready for a second look.

oguzkocer commented:
@0nko I updated all the helper names to include a verb in 07e79ab08a4ebf51722c99b86372936d65bec9b0. I also changed a string in 02a87ed1ec77d6d45a069cc2992789a34d768c90 for [this](https://github.com/wordpress-mobile/WordPress-Android/pull/7829#issuecomment-394301233). It should be ready for another look!

0nko commented:
That looks perfect. Thanks @oguzkocer!

:shipit:

oguzkocer commented:
@0nko Thank **you** for the suggestions and explanation!
